packages/text-editor/src/utils.ts

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright © 2023 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { TiptapCollabProvider } from './provider'
+import * as Y from 'yjs'
+
+type ProviderData = (
+  | {
+    provider: TiptapCollabProvider
+  }
+  | {
+    collaboratorURL: string
+    token: string
+  }
+) & { ydoc?: Y.Doc }
+
+function getProvider (documentId: string, providerData: ProviderData, initialContentId?: string): TiptapCollabProvider {
+  if (!('provider' in providerData)) {
+    return new TiptapCollabProvider({
```

Comment:
Do we need to destroy the created provider?

Reply:
Fixed, thanks. I haven't used new `Disposable` TS 5.2 feature as it's not working flawlessly out-of-the-box.